onnxruntime/core/graph/model.cc

```diff
@@ -176,9 +184,14 @@ Model::Model(ModelProto&& model_proto, const PathString& model_path,
     }
   }
 
+  std::unordered_map<std::string, const ONNX_NAMESPACE::FunctionProto*> model_local_functions;
+  for (auto& func : model_proto_.functions()) {
+    model_local_functions[func.domain() + ":" + func.name()] = &func;
+  }
```

Comment:
should this logic to create a key based on domain:name be in a helper func so it's easy to see all the places that use it?